### PR TITLE
Add string descriptions for all LocalizedErrors (#23)

### PR DESCRIPTION
**Pull request by japanese-goblinn**
# Description

Add string descriptions for all LocalizedErrors 

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation comments
- [x] My changes generate no new warnings

### CODE REVIEW

File: Sources/Client/Views/Settings/UpdateView.swift
Comment (stackotter) [inline Sources/Client/Views/Settings/UpdateView.swift:22]: Typo (wownload)

File: Sources/Client/Views/Settings/UpdateView.swift
Comment (stackotter) [inline Sources/Client/Views/Settings/UpdateView.swift:20]: Please end all errors with a full stop for consistency. It's my mistake if I've accidentally forgotten a full stop in any of the documentation comments (which I have done plenty of times 😅)

**Review by stackotter — CHANGES_REQUESTED** [review]
Looking good so far! Thanks for taking up this task, it'll really improve the overall user experience of the client when things go wrong. There are also a bunch more error types in the Core package that get displayed to users (if you're using Xcode I think it might display it in the dependencies section even though it's all part of the same repository which is a bit confusing). Would you be able to add descriptions for those too please?

**Comment by japanese-goblinn** [comment]
> Looking good so far! Thanks for taking up this task, it'll really improve the overall user experience of the client when things go wrong. There are also a bunch more error types in the Core package that get displayed to users (if you're using Xcode I think it might display it in the dependencies section even though it's all part of the same repository which is a bit confusing). Would you be able to add descriptions for those too please?

yep, of course. i have a small vacation for a couple of days, after it i'll fix everything and add descriptions to errors you mentioned. thanks for contribution possibly, looking forward to make some more after finishing this one!

**Comment by stackotter** [comment]
> yep, of course. i have a small vacation for a couple of days, after it i'll fix everything and add descriptions to errors you mentioned. thanks for contribution possibility, looking forward to make some more after finishing this one!

Awesome, have fun on your vacation!

**Comment by japanese-goblinn** [comment]
Hey, @stackotter. I've finished PR. Changes listed below:
- Add period at the end of error description, as you suggested above.
- Use a multi-line string for errors that looks more convenient to be described in multi-line way. If you think  that some of them (or any other) should be rephrased - let me know. 
- Remove unused errors and code.
- Change some code here and there other code to be more swifty and readable
- Refactor some `somethingId` to `somethingID`. Due to Swift API design guidelines **acronyms should be uppercase** if not in the beginning of variable name ([check](https://www.swift.org/documentation/api-design-guidelines/#follow-case-conventions) more details in `Follow case conventions` section)

File: Sources/Core/Sources/Network/Protocol/Packets/ClientboundPacket.swift
Comment (stackotter) [inline Sources/Core/Sources/Network/Protocol/Packets/ClientboundPacket.swift:6]: I was gonna say that I prefer `Id` (which I do because Id is just short for Identifier and it isn't an acronym). But I just checked and Apple does seem to use `ID`, so I guess we can go with that as long as it is updated everywhere that I use `Id` so that it's consistent.

File: Sources/Core/Sources/Network/Protocol/Packets/Play/Clientbound/EntityAttributesPacket.swift
Comment (stackotter) [reply Sources/Core/Sources/Network/Protocol/Packets/Play/Clientbound/EntityAttributesPacket.swift:15]: nvm, I see what you've done, wherever you use a colon followed by an inserted value you don't use a full stop (I just flagged it because it was a one-line message without a full stop but your way is more consistent). That's good 👍

**Review by stackotter — COMMENTED** [review]
I really like the error style you have gone with - it's clear, concise and has enough context. I'm just going to look into the `Id` vs `ID` thing and find how many places I've used `Id`.

**Comment by stackotter** [comment]
I think there are around 350 occurrences of me using `Id` in the codebase, and most of the dependencies that I use have also gone with `Id` instead of `ID`, so I think it's best that we stick to just using `Id` for consistency. I also personally prefer `Id` over `ID`.

File: Sources/Core/Sources/Network/Protocol/Packets/ClientboundPacket.swift
Comment (japanese-goblinn) [reply Sources/Core/Sources/Network/Protocol/Packets/ClientboundPacket.swift:6]: Oh, well, you right, acronym is a subset of abbreviation. I guest ID is a little bit special due to wiki https://en.wikipedia.org/wiki/ID and people using it as Id and as ID but I prefer an Apple way with ID.
I saw what you wrote about ID and will revert it to Id

**Comment by japanese-goblinn** [comment]
@stackotter I fixed the ID's and also noticed that protobuf is prefering `ID` instead of `Id`. I guess that can't be fixed for now, because i don't see any configuration of this.

**Review by stackotter — CHANGES_REQUESTED** [review]
Sorry, I just noticed one more thing (that I forgot to say last time). Your case blocks are at the same indent level as your switch blocks, but the Delta Client code style is to indent the case statements (to give more visual separation). It's possible to tweak your Xcode settings to make that happen automatically in future if it's an annoyance to you. I swear that's actually the last change now 😅

**Comment by japanese-goblinn** [comment]
@stackotter here you go

File: Sources/Core/Sources/Account/Mojang/MojangAPI.swift
Comment (stackotter) [inline Sources/Core/Sources/Account/Mojang/MojangAPI.swift:11]: Indentation issue here

**Review by stackotter — CHANGES_REQUESTED** [review]
Just one indentation issue, then we're good to merge :)

File: Sources/Core/Sources/Account/Mojang/MojangAPI.swift
Comment (japanese-goblinn) [reply Sources/Core/Sources/Account/Mojang/MojangAPI.swift]: Sorry, just not filling focused today 🥲

**Review by stackotter — APPROVED** [review]
Looks good to me now 👍 thank you so much for completing this issue. It has been open for so long and I just never got around to doing it. This is going to improve the user experience (and even developer experience) of the client so much!

**Comment by stackotter** [comment]
Just going to wait for the build to succeed before I merge. It hopefully should within 10 minutes or so.

**Comment by japanese-goblinn** [comment]
@stackotter thanks for the possibility to contribute. Looking forward to create new exiting (or not) PR's 😅

**Comment by stackotter** [comment]
> thanks for the possibility to contribute

No, thank **you** for contributing! :) It's always good to have another contributor on board

**Pull request merged**
